flutter_bird_app/lib/controller/authentication_service.dart

```diff
@@ -26,69 +26,117 @@ abstract class AuthenticationService {
+  final Function() onAuthStatusChanged;
+  WalletProvider? _lastUsedWallet;
+
+  static const String projectId = String.fromEnvironment('WALLET_CONNECT_PROJECT_ID');
+
+  AuthenticationServiceImpl({
+    required this.isInLiff,
+    required this.operatingChain,
+    required this.onAuthStatusChanged,
+  });
+
+  List<WalletProvider> _availableWallets = [];
   Web3App? _connector;
-  Function() onAuthStatusChanged;
+  bool _isInitialized = false;
 
   @override
-  String get operatingChainName => operatingChain == 1001 ? 'Klaytn Testnet' : 'Chain $operatingChain';
+  List<WalletProvider> get availableWallets => _availableWallets;
 
   @override
   Account? get authenticatedAccount => _authenticatedAccount;
   Account? _authenticatedAccount;
 
   @override
-  bool get isOnOperatingChain => currentChain == operatingChain;
-
-  SessionData? get currentSession => _connector?.sessions.getAll().firstOrNull;
+  String get operatingChainName => operatingChain == 1001 ? 'Klaytn Testnet' : 'Chain $operatingChain';
 
-  int? get currentChain => int.tryParse(currentSession?.namespaces['eip155']?.accounts.first.split(':')[1] ?? '');
+  @override
+  bool get isOnOperatingChain => currentChain == operatingChain;
 
   @override
   bool get isAuthenticated => isConnected && authenticatedAccount != null;
 
+  @override
+  String? get webQrData => _webQrData;
+  String? _webQrData;
+
+  WalletProvider? get lastUsedWallet => _lastUsedWallet;
+
+  SessionData? get currentSession => _connector?.sessions.getAll().firstOrNull;
   bool get isConnected => currentSession != null;
 
-  // The data to display in a QR Code for connections on Desktop / Browser.
+  int? get currentChain => int.tryParse(currentSession?.namespaces['eip155']?.accounts.first.split(':')[1] ?? '');
+
   @override
-  String? webQrData;
+  Future<void> initialize(bool isInLiff) async {
+    if (_isInitialized) {
+      return;
+    }
+    
+    await _createConnector();
+    await _clearSessions();
 
-  AuthenticationServiceImpl({
-    required this.operatingChain,
-    required this.onAuthStatusChanged,
-  }) {
-    if (kIsWeb) {
-      requestAuthentication();
+    if (!kIsWeb || isInLiff) {
+      await _loadWallets();
     } else {
-      _loadWallets();
+      print('AuthenticationServiceImpl: Skipping wallet loading for Web');
     }
+    
+    _isInitialized = true;
   }
 
-  /// Loads all WalletConnect compatible wallets
-  _loadWallets() async {
-    final walletResponse = await http.get(Uri.parse('https://registry.walletconnect.org/data/wallets.json'));
-    final walletData = json.decode(walletResponse.body);
-    availableWallets = walletData.entries.map<WalletProvider>((data) => WalletProvider.fromJson(data.value)).toList();
+  // Update to support WalletConnect v2
+  Future<void> _loadWallets() async {
+    try {
+      final response = await http.get(
+        Uri.parse('https://explorer-api.walletconnect.com/v3/wallets?projectId=${projectId}&entries=5&page=1'),
```

Comment:
Does the change in the WalletProvider model correspond to the update in this URL?

Reply:
I've updated the WallertProvider model to a format that can be retrieved by this URL. Does this answer your question?

Reply:
Yes, it does. Thank you!